crates/turbopack-dev/src/chunking_context.rs

```diff
@@ -367,16 +447,60 @@ impl EcmascriptChunkingContext for DevChunkingContext {
     }
 }
 
-#[turbo_tasks::value_impl]
-impl EvaluateChunkingContext for DevChunkingContext {
-    #[turbo_tasks::function]
-    fn evaluate_chunk(
-        self_vc: DevChunkingContextVc,
-        entry_chunk: ChunkVc,
-        other_chunks: AssetsVc,
-        evaluatable_assets: EvaluatableAssetsVc,
-    ) -> AssetVc {
-        EcmascriptDevEvaluateChunkVc::new(self_vc, entry_chunk, other_chunks, evaluatable_assets)
-            .into()
+async fn get_optimized_parallel_chunks<I>(entries: I) -> Result<ChunksVc>
+where
+    I: IntoIterator<Item = ChunkVc>,
+{
+    let chunks: Vec<_> = GraphTraversal::<SkipDuplicates<ReverseTopological<_>, _>>::visit(
+        entries,
+        get_chunk_children,
+    )
+    .await
+    .completed()?
+    .into_inner()
+    .into_iter()
+    .collect();
+
+    let chunks = ChunksVc::cell(chunks);
+    let chunks = optimize(chunks);
+
+    Ok(chunks)
+}
+
+/// Computes the list of all chunk children of a given chunk.
+async fn get_chunk_children(parent: ChunkVc) -> Result<impl Iterator<Item = ChunkVc> + Send> {
+    Ok(parent
+        .references()
+        .await?
+        .iter()
+        .copied()
+        .map(reference_to_chunks)
+        .try_join()
+        .await?
+        .into_iter()
+        .flatten())
```

Comment:
The `Chunk` trait should have a `parallel_chunks()` method to expose parallel chunks. `references()` should no longer contain parallel chunks